### ISSUE TITLE
[Feature Request] Stepper should have prop to prevent clicking beyond last reached step.

### ISSUE THREAD

**Issue by LucasLemanowicz** [mantinedev/mantine#555]
### Usage example, including component/hook API

For certain workflows you'd want your customers to be able to navigate freely throughout the steps they've already accessed, but not further ahead. It would be nice to have that configurable via a prop:

```typescript
const [page, setPage] = useState<number>(0);

// ...

<Stepper active={page} onStepClick={setPage} preventClickAhead={true}>
    <Stepper.Step>
        Welcome Instructions
    </Stepper.Step>
    <Stepper.Step>
        First Form Page
    </Stepper.Step>
    <Stepper.Step>
        Second Form Page
    </Stepper.Step>
    <Stepper.Step>
        Confirmation
    </Stepper.Step>
</Stepper>

```

In the example above, if the customer reached the First Form Page, they would be able to click to jump between the Welcome Instructions step and the First Form Page step, but not yet to the Second Form Page. Only once they complete certain actions (for example filling out the form) and click on the Next Page button at the bottom (separately rendered / implemented) would they be able to move on.

### Possible implementation – describe how the feature can be implemented

A possible implementation of this:
1. The stepper would store either the latest step reached, or a list of all steps that have been accessed
2. Modify the `allowStepClick` to something like `allowStepClick={typeof onStepClick === 'function' && (preventClickAhead ? accessedSteps.indexOf(index) != -1)} : true)`

**Comment by rtivital** [comment]
Released in 3.6.0
